services/src/main/java/org/keycloak/organization/admin/resource/OrganizationIdentityProvidersResource.java

```diff
@@ -0,0 +1,139 @@
+        this.session = session;
+        this.realm = session == null ? null : session.getContext().getRealm();
+        this.organizationProvider = session == null ? null : session.getProvider(OrganizationProvider.class);
+        this.organization = organization;
+        this.auth = auth;
+        this.adminEvent = adminEvent;
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response addIdentityProvider(IdentityProviderRepresentation providerRep) {
+        auth.realm().requireManageRealm();
+
+        Response response = new IdentityProvidersResource(realm, session, auth, adminEvent).create(providerRep);
+
+        if (Status.CREATED.getStatusCode() == response.getStatus()) {
+            try {
+                IdentityProviderModel identityProvider = realm.getIdentityProviderByAlias(providerRep.getAlias());
+
+                if (organizationProvider.addIdentityProvider(organization, identityProvider)) {
+                    return response;
+                }
+
+                throw ErrorResponse.error("Identity provider already associated to the organization", Status.BAD_REQUEST);
+            } catch (ModelException me) {
+                throw ErrorResponse.error(me.getMessage(), Status.BAD_REQUEST);
+            }
+        }
+
+        return response;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<IdentityProviderRepresentation> getIdentityProviders() {
+        auth.realm().requireManageRealm();
+        return organization.getIdentityProviders()
+                .map(this::toRepresentation)
+                .toList();
+    }
+
+    @Path("{alias}")
+    public IdentityProviderResource getIdentityProvider(@PathParam("alias") String alias) {
+        IdentityProviderModel broker = realm.getIdentityProviderByAlias(alias);
+        return new IdentityProviderResource(auth, realm, session, broker, adminEvent) {
+            @Override
+            public Response delete() {
+                Response response = super.delete();
+
+                if (organizationProvider.removeIdentityProvider(organization, broker)) {
+                    return response;
+                }
+
+                throw ErrorResponse.error("Identity provider not associated with the organization", Status.BAD_REQUEST);
+            }
+
+            @Override
+            public Response update(IdentityProviderRepresentation providerRep) {
+                if (organization.getIdentityProviders().noneMatch(model -> model.getInternalId().equals(providerRep.getInternalId()) || model.getAlias().equals(providerRep.getAlias()))) {
+                    return Response.status(Status.NOT_FOUND).build();
+                }
+                String domain = providerRep.getConfig().get(OrganizationModel.ORGANIZATION_DOMAIN_ATTRIBUTE);
+
+                if (domain != null && organization.getDomains().map(OrganizationDomainModel::getName).noneMatch(domain::equals)) {
```

Comment:
Feel free to disregard, but curious if you'd rather use `StringUtils.isNotBlank(domain)` here instead? java string comparison is a bit quirky

Reply:
In this case, we are not comparing elements in the string that is why I'm using `domain::equals`. As long it is not null, works fine.